praw/__init__.py

```diff
@@ -1731,18 +1731,28 @@ def set_stylesheet(self, subreddit, stylesheet):
         return self.request_json(self.config['subreddit_css'], data=data)
 
     @decorators.restrict_access(scope='modconfig')
-    def upload_image(self, subreddit, image_path, name=None, header=False):
+    def upload_image(self, subreddit, image_path, name=None,
+                     header=False, upload_as=None):
         """Upload an image to the subreddit.
 
         :param image_path: A path to the jpg or png image you want to upload.
         :param name: The name to provide the image. When None the name will be
             filename less any extension.
         :param header: When True, upload the image as the subreddit header.
+        :param upload_as: Must be `'jpg'`, `'png'` or `None`. When None, this
+            will match the format of the image itself. In all cases where both
+            this value and the image format is not png, reddit will also
+            convert  the image mode to RGBA. reddit optimizes the image
+            according to this value.
         :returns: A link to the uploaded image. Raises an exception otherwise.
 
         """
         if name and header:
             raise TypeError('Both name and header cannot be set.')
+        if upload_as not in (None, 'png', 'jpg', 'jpeg'):
+            raise TypeError("upload_as must be 'jpg', 'png', or None.")
+        elif upload_as == "jpeg":
+            upload_as = "jpg"
         image_type = None
         # Verify image is a jpeg or png and meets size requirements
         with open(image_path, 'rb') as image:
```

Comment:
Can you extract this image_type calculation out to it's own private function (e.g., `_compute_image_type(image_path)`) and then only call it in the case that `upload_as` is `None`?